samples/subsys/usb/cdc_acm/src/main.c

```diff
@@ -41,12 +41,27 @@ static void interrupt_handler(struct device *dev)
 
 static void write_data(struct device *dev, const char *buf, int len)
 {
+	int sent;
+	u32_t transfer_len;
 	uart_irq_tx_enable(dev);
 
-	data_transmitted = false;
-	uart_fifo_fill(dev, (const u8_t *)buf, len);
-	while (data_transmitted == false)
-		;
+	while (len) {
+		data_transmitted = false;
+		/* FIXME: An issue observed when USB TX Endpoint is written
+		 * more than once (greater than 4 Bytes of data is written)
+		 * (Jira 2074). So restricting writing 4 bytes at a time into
+		 * USB TX Endpoing.
+		 */
+		transfer_len = len > sizeof(u32_t) ? sizeof(u32_t) : len;
+		sent = uart_fifo_fill(dev, (const u8_t *)buf, transfer_len);
```

Comment:
Should be no need for the loop in cdc_acm_fifo_fill(). The interface for uart_fifo_fill() is that it writes as much as it safely can (i.e. some number of bytes between 0 and len) and then returns the number of bytes it managed to get out.

Thus the loop should end up in the example (as you had it before). It is only the knowledge that "it is unsafe to write more than 4 bytes" that needs to be in cdc_acm_fifo_fill().

On other words you just add the FIXME comment explaining the problem (and on what platform it is observed) and `len = (len > 4 ? 4 : len)` to the top of cdc_acm_fifo_fill().